sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/checkin/SentryCheckInAdvice.java

```diff
@@ -51,7 +55,20 @@ public Object invoke(final @NotNull MethodInvocation invocation) throws Throwabl
     }
 
     final boolean isHeartbeatOnly = checkInAnnotation.heartbeat();
-    final @Nullable String monitorSlug = checkInAnnotation.value();
+
+    @Nullable String monitorSlug = checkInAnnotation.value();
+
+    if (resolver != null) {
+      try {
+        monitorSlug = resolver.resolveStringValue(checkInAnnotation.value());
+      } catch (Throwable e) {
+        hub.getOptions()
```

Comment:
```suggestion
        // When resolving fails, we fall back to the original string which may contain unresolved expressions. Testing shows this can also happen if properties cannot be resolved. Sentry should alert the user about missed checkins in this case since the monitor slug won't match what is configured in Sentry.
        hub.getOptions()
```

Should we include a comment like this ^ ? Needs some formatting, just want to write this down so we don't forget.